docs/amended-alias-resolution.md

```diff
@@ -0,0 +1,165 @@
+This means that any unprefixed string is always treated as an unaliased path.
+```luau
+-- Requires ./libs/dependency.luau
+require("libs/dependency")
+
+-- Requires ./libs/dependency.luau
+require("./libs/dependency")
+
+-- Requires /My/Libraries/Directory/dependency.luau
+require("@libs/dependency")
+```
+
+The main drawback of this approach is that projects with many _external_ dependencies will have many `@` symbols appearing within require statements.
+
+This is similar to [Lune's approach](https://lune-org.github.io/docs/getting-started/2-introduction/8-modules#file-require-statements) and is identical to the approach currently described in [require-by-string-aliases.md](require-by-string-aliases.md).
+
+#### (1b) Make relative paths explicit with either `./` or `../` prefix
+
+We make usage of either the `./` or `../` prefix necessary in relative require statements.
+This means that any unprefixed string is always treated as an aliased path.
+```luau
+-- Requires /My/Libraries/Directory/dependency.luau
+require("libs/dependency")
+
+-- Requires ./libs/dependency.luau
+require("./libs/dependency")
+
+-- Requires /My/Libraries/Directory/dependency.luau
+require("@libs/dependency")
+```
+
+The main drawback of this approach is that projects with many _internal_ dependencies will have many `./` and `../` symbols appearing within require statements.
+
+This is similar to [darklua's approach](https://darklua.com/docs/path-require-mode/).
+
+### (2) Strongly recommend explicit prefixes
+
+We don't make the `@`, `./`, or `../` prefixes necessary, but we strongly recommend them.
+
+This can be accomplished by allowing unprefixed paths but throwing an error if they resolve to multiple files.
+There is a performance cost to this: currently, if we successfully match a given path to a file, we stop searching for other matches.
+If this approach were implemented, we would need to search for at most two matching files in case we need to throw an error.
+In the case of multiple nested directories with their own `.luaurc` files and `paths` arrays, this search could take linear time with respect to the total number of paths defined, in the worst case.
+
+Of course, developers could use the `@`, `./`, or `../` prefixes to explicitly specify a search type and eliminate this performance cost.
+```luau
+-- Error: ./libs/dependency.luau exists and .luaurc defines a distinct "libs" alias
+-- Would require ./libs/dependency.luau if "libs" alias did not exist
+-- Would require /My/Libraries/Directory/dependency.luau if ./libs/dependency.luau did not exist
+require("libs/dependency")
+
+-- Requires ./libs/dependency.luau
+require("./libs/dependency")
+
+-- Requires /My/Libraries/Directory/dependency.luau
+require("@libs/dependency")
+```
+
+This is a compromise between approaches (1a) and (1b) that sacrifices performance for compatibility and readability.
+
+## Alternatives
+
+### All explicit prefixes necessary
```

Comment:
Why is this under Alternatives instead of a design under ongoing consideration? "but it is not backwards compatible with existing code" is something we broke several times already in non-Roblox contexts.

Reply:
Honestly, this is a bit of an untraditional RFC because it's not exactly proposing anything yet, just laying out options. I'll bring in the alternatives into the main design section, and move the rejected ones back to Alternatives once we've all discussed the options.